numba/core/typed_passes.py

```diff
@@ -437,16 +437,45 @@ def run_pass(self, state):
 
 
 @register_pass(mutates_CFG=False, analysis_only=True)
-class IRLegalization(AnalysisPass):
+class NoPythonSupportedFeatureValidation(AnalysisPass):
+    """NoPython Mode check: Validates the IR to ensure that features in use are
+    in a form that is supported"""
 
-    _name = "ir_legalization"
+    _name = "no_python_supported_feature_validation"
```

Comment:
```suggestion
    _name = "nopython_supported_feature_validation"
```

we always refer to it as one word; else it reads like it not doing validation

Reply:
indeed, fixed!